### ISSUE TITLE
TST: ASDF 2.5.0 broke some tests

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#9816]
Our CI is now picking up [ASDF 2.5.0](https://github.com/spacetelescope/asdf/releases/tag/2.5.0), which broke the following tests:

```
____________________________ test_backwards_compat _____________________________
    def test_backwards_compat():
        """
        Make sure that we can continue to read FITS HDUs that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the fits schema from the ASDF Standard,
        since these make no reference to Astropy's own fits definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], fits.HDUList)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'fits/fits', '1.0.0', check)
astropy/io/misc/asdf/tags/fits/tests/test_fits.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/asdf/tags/tests/helpers.py:14: in run_schema_example_test
    uri = asdf.resolver.default_tag_to_url_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
uri = 'tag:stsci.edu:asdf/fits/fits-1.0.0'
    def default_tag_to_url_mapping(uri):
>       warnings.warn("'default_tag_to_url_mapping' is deprecated.", AsdfDeprecationWarning)
E       asdf.exceptions.AsdfDeprecationWarning: 'default_tag_to_url_mapping' is deprecated.
.../asdf/resolver.py:169: AsdfDeprecationWarning
____________________________ test_backwards_compat _____________________________
    def test_backwards_compat():
        """
        Make sure that we can continue to read tables that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the table schema from the ASDF Standard,
        since these make no reference to Astropy's own table definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], table.Table)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'core/table', '1.0.0', check)
astropy/io/misc/asdf/tags/table/tests/test_table.py:238: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/asdf/tags/tests/helpers.py:14: in run_schema_example_test
    uri = asdf.resolver.default_tag_to_url_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
uri = 'tag:stsci.edu:asdf/core/table-1.0.0'
    def default_tag_to_url_mapping(uri):
>       warnings.warn("'default_tag_to_url_mapping' is deprecated.", AsdfDeprecationWarning)
E       asdf.exceptions.AsdfDeprecationWarning: 'default_tag_to_url_mapping' is deprecated.
.../asdf/resolver.py:169: AsdfDeprecationWarning
```

Example log: https://travis-ci.org/astropy/astropy/jobs/629110036

As a result, the failures caused unrelated push events to fail as well (e.g., https://github.com/astropy/astropy/pull/9780#issuecomment-568782904).

**Comment by jdavies-st** [comment]
It looks like astropy CI is turning warnings into errors?  It is just a deprecation warning coming from `asdf`.  It should not be causing an error.

**Comment by jdavies-st** [comment]
The problem is here in `setup.cfg`:
```
filterwarnings =
    error
    ignore:unclosed file:ResourceWarning
    ignore:unclosed <socket:ResourceWarning
    ignore:unclosed <ssl.SSLSocket:ResourceWarning
    ignore:The timefunc function is deprecated
    ignore:numpy.ufunc size changed:RuntimeWarning
    ignore:Using or importing the ABCs from 'collections':DeprecationWarning
    ignore:can't resolve package from __spec__
    ignore:PY_SSIZE_T_CLEAN will be required for '#' formats
    ignore:::astropy.tests.plugins.display
    ignore:::astropy.tests.disable_internet
```
which turns all warnings into errors, except for the ones that are ignored.  This short-circuits how `DeprecationWarnings` are supposed to work.  I.e. give package maintainers time to fix code that depends on deprecated code in dependencies.  The way it is set up now means that any warning from a dependency will fail your CI.  Not good.  Especially if push events depend on your CI passing.

That said, I realize that `AsdfDeprecationWarning` does not subclass `DeprecationWarning`, as it should, so that needs to be fixed in `asdf`.  Though that does not affect this failure in the `astropy` tests.  Filtering warnings into errors as above will cause all warnings generated to fail your tests.

**Comment by bsipocz** [comment]
> The way it is set up now means that any warning from a dependency will fail your CI. Not good. Especially if push events depend on your CI passing.

That is the exact intended behaviour, otherwise it could easily happen to not notice a deprecation in time and deprecated usage then more likely to get into a released versions.  For packages where we test against the dev version this also help to have a super quick feedback loop, so they could revise  the deprecation if necessary.
Anyway,  the usual approach is to have a quick fix if possible when we notice it (like #9819, thanks for it), or pin the version number of the upstream package until we get around to fix stuff here.
